gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+                  'UPDATE_FIREWALL': 'update_network_service_config',
+                  'DELETE_FIREWALL': 'delete_network_service_config'}
+        request_data = self.fo.fake_request_data_fw()
+        with mock.patch.object(sc, 'new_event', return_value='foo') as (
+                                                        mock_sc_event), \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event, \
+            mock.patch.object(rpc_mgr,
+                              '_get_service_agent_instance',
+                              return_value=agent):
+            getattr(rpc_mgr, method[operation])(self.fo.context, request_data)
+
+            mock_sc_event.assert_called_with(id=operation, data=arg_dict)
+            mock_sc_rpc_event.assert_called_with('foo')
+
+    def _test_notifications(self, bulk=False):
+        sc, conf, rpc_mgr = self._get_ConfiguratorRpcManager_object()
+        agent = self._get_GenericConfigEventHandler_object(sc, rpc_mgr)
+        
+        data = "PUT ME IN THE QUEUE!"
+        with mock.patch.object(sc, 'new_event', return_value='foo') as (
+                                                        mock_new_event), \
+            mock.patch.object(sc, 'poll_event') as (mock_poll_event):
+            
+            agent._notification(data)
+
+            mock_new_event.assert_called_with(id='NOTIFICATION_EVENT',
+                                              key='NOTIFICATION_EVENT',
+                                              data=data)
+            mock_poll_event.assert_called_with('foo')
+        
+    def test_configure_routes(self):
+        method = "CONFIGURE_ROUTES"
+        operation = 'create'
+        self._test_generic_event_creation(operation, method)
+        
+    def test_clear_routes(self):
+        method = "CLEAR_ROUTES"
+        operation = 'delete'
+        self._test_generic_event_creation(operation, method)
+
+    def test_configure_interfaces(self):
+        method = "CONFIGURE_INTERFACES"
+        operation = 'create'
+        self._test_generic_event_creation(operation, method)
+        
+    def test_clear_interfaces(self):
+        method = "CLEAR_INTERFACES"
+        operation = 'delete'
+        self._test_generic_event_creation(operation, method)
+        
+    def test_configure_bulk(self):
+        method = "PROCESS_BATCH"
+        operation = 'create'
+        self._test_generic_event_creation(operation, method, True)
+
+    def test_clear_bulk(self):
+        method = "PROCESS_BATCH"
+        operation = 'delete'
+        self._test_generic_event_creation(operation, method, True)
+
+    def test_create_firewall(self):
+        self._test_fw_event_creation('CREATE_FIREWALL')
+        
+    def test_update_firewall(self):
+        self._test_fw_event_creation('UPDATE_FIREWALL')
+
+    def test_delete_firewall(self):
+        self._test_fw_event_creation('DELETE_FIREWALL')
+
+    def test_get_notifications_generic(self):
```

Comment:
There are notifications for firewall also right? However, the method code is same. Also, since this _notification() is moved to base class, we need to rename this.